lib/shoryuken/fetcher.rb

```diff
@@ -44,6 +48,36 @@ def fetch_with_auto_retry(max_attempts)
       end
     end
 
+    def regular_fetch(queue, limit)
+      sqs_msgs = Array(receive_messages(queue, limit))
+      logger.debug { "Found #{sqs_msgs.size} messages for #{queue.name}" } unless sqs_msgs.empty?
+      sqs_msgs
+    end
+
+    def fetch_with_batch_options(queue, limit, batch_max_size, batch_timeout)
+      batch = Shoryuken::MessageBatch.new(max_size: batch_max_size, timeout: batch_timeout)
+      loop do
+        check_visibility_timeout(queue, batch.timeout)
```

Comment:
Does this need to be inside of the loop? It seems the first time it gets true. It gets memorized and always return.

Reply:
Yes, the check can be performed only once outside of the loop. I decided to put it inside the loop to be more aggressive with the warning log, but eventually, the loop will end so we'll come back on the check multiple times anyway. Thanks for the suggestion.